dask_bigquery/core.py

```diff
@@ -317,7 +318,12 @@ def to_gbq(
         df.to_parquet(path, **parquet_kwargs_used)
 
         with bigquery_client(project_id, credentials=credentials) as client:
-            target_dataset = client.create_dataset(dataset_id, exists_ok=True)
+            # first try getting the dataset, in case user has read but not create
+            # permissions. If it doesn't exist, we must create.
+            try:
```

Comment:
In the tests, I'm seeing that we are testing that we can write to an existing data set and also to create a new one. But I'm missing a test that will check that the right path is chosen based on what I read in this try/except. 

Something that checks when there is an `exception.NotFound` then the dataset will be created. Would it be possible to trigger this exception easily with the creds the bot has? @j-bennet any thoughts?

Reply:
We already (non-explicitly) trigger this use case whenever `write_dataset` fixture is used:

https://github.com/coiled/dask-bigquery/blob/5521666aa5f95b31c2cda6bd150c372ff529fc13/dask_bigquery/tests/test_core.py#L143-L157

Notice how the fixture does not create a dataset, it just provides you a unique id for a new dataset, and afterwards, does the cleanup.

it would still be good to test with a set of credentials for a lower-permission user though. I opened an issue for that: https://github.com/coiled/dask-bigquery/issues/76.